src/checkers/inference/model/VariableSlot.java

```diff
@@ -46,7 +46,7 @@ public VariableSlot(AnnotationLocation location, int id) {
     }
 
     // Slots this variable has been merged to.
-    private final Set<CombVariableSlot> mergedToSlots = new HashSet<CombVariableSlot>();
+    private final Set<TernaryVariableSlot> mergedToSlots = new HashSet<TernaryVariableSlot>();
```

Comment:
Similar to in `DefaultSlotManager`, `mergedToSlots` is for storing `LeastUpperBoundSlot`. Why generalizing it to `TernaryVariableSlot`? I assume `TernaryVariableSlot` is not to model LeastUpperBoundSlot`?